fenix/app/src/main/java/org/mozilla/fenix/messaging/state/MessagingMiddleware.kt

```diff
@@ -77,7 +80,12 @@ class MessagingMiddleware(
                 consumeMessageToShowIfNeeded(context, oldMessage)
                 removeMessage(context, oldMessage)
             }
-            context.store.dispatch(UpdateMessages(newMessages))
+
+            MainScope().launch {
+                context.store.dispatch(UpdateMessages(newMessages))
+            }
```

Comment:
Could you clarify why do we need to switch thread?

Reply:
Store was created on the main thread so the dispatch [Middleware.kt](https://searchfox.org/mozilla-mobile/rev/21548505501d6cee5ce2aac3244821269bc7969d/firefox-android/android-components/components/lib/state/src/main/java/mozilla/components/lib/state/Middleware.kt#42) to be on main thread too.